mon-pix/app/controllers/authenticated/user-account/language.js

```diff
@@ -3,14 +3,18 @@ import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 
 export default class UserAccountPersonalInformationController extends Controller {
+  @service currentUser;
+  @service dayjs;
   @service intl;
   @service currentDomain;
-  @service location;
 
   @action
-  async onLanguageChange(value) {
+  async onLanguageChange(language) {
     if (!this.currentDomain.isFranceDomain) {
-      this.location.replace(`/mon-compte/langue?lang=${value}`);
+      this.intl.setLocale([language]);
```

Comment:
question : 

- tu passes [language] à setLocal ? C'est voulu ? 

Il y beaucoup de répétition de ce bout de code (on passe un tableau à **setLocale** mais pas toujours)

 this.intl.setLocale([language]);
 this.dayjs.setLocale(language);

On pourrait peut-être commencer à penser à un refacto  dans le service Locale ? Qu'en penses tu ?

Reply:
ps : peut-être pas dans ce ticket

Reply:
Hmm, en effet en regardant la [doc](https://ember-intl.github.io/ember-intl/docs/api/modules/ember-intl/test-support/set-locale) sur setLocale, on peut passer une chaîne de caractère. Je corrige ça 😸   

Concernant la refacto, bonne remarque ! 
Par contre, je préfère faire ça dans un autre ticket vu que ça touche toutes les apps.
Ticket crée => PIX-7922